text/0006-sampling.md

```diff
@@ -0,0 +1,300 @@
+# Sampling API
+
+*Status: proposed*
+
+## TL;DR
+This section tries to summarize all the changes proposed in this RFC:
+ 1. Move the `Sampler` interface from the API to SDK package. Apply some minor changes to the
+ `Sampler` API.
+ 1. Add a new `SamplerHint` concept to the API package.
+ 1. Add capability to record `Attributes` that can be used for sampling decision during the `Span`
```

Comment:
I don't understand why we would call these synthetic things "Attributes".  They're not given by the programmer, they're generated by the SDK. Can we record these key:values in a way that marks them clearly as having been injected by the implementation? They're not the user's attributes.

Reply:
They are the user's attributes, and can/should be supplied by the programmer before being used for sampling.

Reply:
Oh, that wasn't clear to me. I thought we already had such a requirement, since the OpenTracing start method includes a list of attributes. I would state this simply as a requirement that the user be allowed to specify attributes at start time, practically unrelated to sampling. Thanks.